JIP-5.md

```diff
@@ -0,0 +1,77 @@
+# JIP-5: Secret key derivation
+
+This JIP proposes a standard method for deriving a set of validator secret keys from a 32-byte
+seed.
+
+## Derivation method
+
+Given a 32-byte seed `seed`, a set of secret keys is derived as follows:
+
+    ed25519_secret = blake2b("jam_val_key_ed25519" ++ seed)
+    bandersnatch_secret = blake2b("jam_val_key_bandersnatch" ++ seed)
+    bandersnatch_scalar = decode_as_le(sha512(bandersnatch_secret)) % r
+
+Where:
+
+- The strings are ASCII-encoded with no terminator.
+- `++` indicates concatenation.
+- `blake2b` is the BLAKE2b hash function with 32-byte output.
+- `sha512` is the SHA-512 hash function.
+- `decode_as_le` is a function that decodes the input as a little-endian unsigned integer.
+- `r` is the order of the prime subgroup of the Bandersnatch elliptic curve.
+- `ed25519_secret` is a 32-byte Ed25519 secret key as defined in
+  <https://ed25519.cr.yp.to/ed25519-20110926.pdf>.
+- `bandersnatch_scalar` is a secret key scalar as defined in
+  <https://github.com/davxy/bandersnatch-vrf-spec/blob/main/specification.pdf>.
+
+## Trivial seeds
+
+We define the following function for deriving a 32-byte seed from a 32-bit unsigned integer:
+
+    trivial_seed(i) = repeat_8_times(encode_as_32bit_le(i))
+
+Such seeds should be used for testing purposes only.
+
+## Test vectors
+
+    seed = trivial_seed(0) = 0000000000000000000000000000000000000000000000000000000000000000
+    ed25519_secret = 996542becdf1e78278dc795679c825faca2e9ed2bf101bf3c4a236d3ed79cf59
+    ed25519_public = 4418fb8c85bb3985394a8c2756d3643457ce614546202a2f50b093d762499ace
+    bandersnatch_secret = 007596986419e027e65499cc87027a236bf4a78b5e8bd7f675759d73e7a9c799
+    bandersnatch_public = ff71c6c03ff88adb5ed52c9681de1629a54e702fc14729f6b50d2f0a76f185b3
+
+    seed = trivial_seed(1) = 0100000001000000010000000100000001000000010000000100000001000000
+    ed25519_secret = b81e308145d97464d2bc92d35d227a9e62241a16451af6da5053e309be4f91d7
+    ed25519_public = ad93247bd01307550ec7acd757ce6fb805fcf73db364063265b30a949e90d933
+    bandersnatch_secret = 12ca375c9242101c99ad5fafe8997411f112ae10e0e5b7c4589e107c433700ac
```

Comment:
Could we also add the `bandersnatch_scalar` here or is it implementation specific?  
The BS lib that I use (the one that probably everybody uses) returns some other value than this, but produces the expected Public Key with this input.  

I assume that it returns the scalar - could help to have it included here for comparison in that case.

Reply:
> The BS lib that I use (the one that probably everybody uses) returns some other value than this, but produces the expected Public Key with this input.

LOL you mean ark-vrf?